### PR TITLE
Feature/7 mock offset and counts per rotation support

### LINKED ISSUE #7: Mock Offset and Counts Per Rotation Support
To support testing with more dynamic encoder settings, the WRoboclaw library should include support for per-encoder ROS parameters for the following:
* `offset`: This should be the starting value of the mock encoder before motion occurs
* `counts_per_rotation`: This is the maximum value of the encoder.  If the count would go over this value or under 0, then it should 'wrap around' to the other bound.

The real Roboclaw objects should not implement these behaviors, since they read them from hardware directly.

### PR DESCRIPTION
**Pull request by JackZautner**
Added support for setting counts per rotation and offsets for roboclaws. The real roboclaws will just read from the hardware but the mock ones will be able to read from a file, which will include the parameters. Important to note: Only the logic to do this has been implemented so far, I have not modified the parameter file. [Fixed issue #7]

### CODE REVIEW

File: nodes/drive_roboclaw.py
Comment (bennowotny) [inline nodes/drive_roboclaw.py:188]: Don't think we want to keep this in a finalized version...

File: src/wroboclaw/roboclaw/serial_interface.py
Comment (bennowotny) [inline src/wroboclaw/roboclaw/serial_interface.py:206]: Maybe comment to signal why these methods are do-nothings.

Comment (bennowotny) [reply src/wroboclaw/roboclaw/serial_interface.py:206]: Imagine someone else coming by in 2-3 years...
"These methods don't have any behavior; I'll just remove them!"
And the woe that befalls them afterwards.

File: src/wroboclaw/util/mock.py
Comment (bennowotny) [inline src/wroboclaw/util/mock.py:53]: If these methods are supposed to be publicly accessible (and they are, you use them that way, that's ok), they should not use the 'private' method convention of starting with an underscore.

File: nodes/drive_roboclaw.py
Comment (bennowotny) [inline nodes/drive_roboclaw.py:13]: Should this go to the `util` package somewhere?  I don't know for certain, but now it's defined in two places (mock.py:8), even though it's the exact same constant.

**Review by bennowotny — CHANGES_REQUESTED** [review]
Some small fixes, noted in comments.

**Review by co-li — APPROVED** [review]
Looks good to me, previous comments have been addressed.

File: nodes/drive_roboclaw.py
Comment (bennowotny) [inline nodes/drive_roboclaw.py:12]: This should probably go to `util.py` based on where it's used.

**Review by bennowotny — CHANGES_REQUESTED** [review]

File: nodes/drive_roboclaw.py
Comment (bennowotny) [reply nodes/drive_roboclaw.py]: Complete as of cb491a0.

File: src/wroboclaw/util/mock.py
Comment (bennowotny) [reply src/wroboclaw/util/mock.py]: Complete as of cb491a0.

File: src/wroboclaw/roboclaw/serial_interface.py
Comment (bennowotny) [reply src/wroboclaw/roboclaw/serial_interface.py]: Complete as of cb491a0.

File: nodes/drive_roboclaw.py
Comment (bennowotny) [reply nodes/drive_roboclaw.py]: Complete as of cb491a0.

**Review by bennowotny — APPROVED** [review]
Looks good.

File: nodes/drive_roboclaw.py
Comment (bennowotny) [reply nodes/drive_roboclaw.py]: Resolved in aac6107.

**Comment by bennowotny** [comment]
This still needs a sync merge before this can merge.  That can be done with the 'resolve conflicts' button, or, while in this branch:
`git pull origin master`
Which will cause a merge for all changes that have happened to the main branch while this was being developed.

**Comment by JackZautner** [comment]
I resolved the merge conflict, not sure if there is anything we have to do before squash merging.

**Comment by bennowotny** [comment]
@JackZautner If it's tested and it works, then it should be good to merge now.

When you're ready, click the merge button (should be green and say "squash and merge") to merge back to wroboclaw master.

**Pull request merged**
